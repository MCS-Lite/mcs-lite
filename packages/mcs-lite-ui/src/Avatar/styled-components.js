// @flow
import * as React from 'react';
import styled from 'styled-components';

const Wrapper: React.ComponentType<*> = styled.div`
  img,
  svg {
    display: inline-block;
    border-radius: 50%;
    border: 1px solid ${props => props.theme.color.grayBase};
    padding: 1px;
    background: ${props => props.theme.color.white};
    box-sizing: border-box;
    object-fit: cover;
  }
`;

export default Wrapper;
