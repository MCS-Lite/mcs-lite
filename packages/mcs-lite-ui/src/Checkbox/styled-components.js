// @flow
import * as React from 'react';
import styled from 'styled-components';
import { darken3 } from 'mcs-lite-theme';

export const Container: React.ComponentType<{
  size?: number,
  kind?: string,
  checked: boolean,
}> = styled.div`
  line-height: 1em;
  font-size: 12px;
  border-radius: 2px;
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  box-sizing: border-box;
  user-select: none;
  cursor: pointer;
  transition: background-color 0.5s cubic-bezier(0.23, 1, 0.32, 1),
    border 0.3s cubic-bezier(0.23, 1, 0.32, 1);
  color: ${props => {
    if (props.checked) {
      return props.kind === 'default'
        ? props.theme.color.grayBase
        : props.theme.color.white;
    }
    return 'transparent';
  }};
  background-color: ${props =>
    props.checked ? props.theme.color[props.kind] : props.theme.color.white};
  border: 1px solid
    ${props =>
      props.checked
        ? darken3(props.theme.color[props.kind])
        : props.theme.color.grayDark};
`;

export const Content: React.ComponentType<{ checked: boolean }> = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  transform: scale(${props => (props.checked ? 1 : 0)});
  transform-origin: 50% 50%;
  transition: transform 0.5s cubic-bezier(0.23, 1, 0.32, 1);

  svg {
    fill: currentColor;
  }
`;
