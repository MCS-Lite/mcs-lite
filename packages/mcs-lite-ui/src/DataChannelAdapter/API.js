const DATA_CHANNELS = [
  {
    id: 'Integer Control id',
    type: 'INTEGER_CONTROL',
    values: { value: 0 },
    format: {
      unit: 'ampere',
    },
  },
  {
    id: 'Integer Display id',
    type: 'INTEGER_DISPLAY',
    values: { value: 0 },
    format: {
      unit: 'ampere',
    },
  },
  {
    id: 'Float Control id',
    type: 'FLOAT_CONTROL',
    values: {},
    format: {
      unit: 'ampere',
    },
  },
  {
    id: 'Float Display id',
    type: 'FLOAT_DISPLAY',
    values: {},
    format: {
      unit: 'ampere',
    },
  },
  {
    id: 'String Control id',
    type: 'STRING_CONTROL',
    values: { value: 'Any string type value.' },
    format: {},
  },
  {
    id: 'String Display id',
    type: 'STRING_DISPLAY',
    values: { value: 'Any string type value.' },
    format: {},
  },
  {
    id: 'Hex Control id',
    type: 'HEX_CONTROL',
    values: { value: 'ABF123' },
    format: {},
  },
  {
    id: 'Hex Display id',
    type: 'HEX_DISPLAY',
    values: { value: 'ABF123' },
    format: {},
  },
  {
    id: 'Switch Control id',
    type: 'SWITCH_CONTROL',
    values: { value: 1 },
    format: {},
  },
  {
    id: 'Switch Display',
    type: 'SWITCH_DISPLAY',
    values: { value: 0 },
    format: {},
  },
  {
    id: 'Category Control id',
    type: 'CATEGORY_CONTROL',
    values: { value: 'v2' },
    format: {
      items: [
        { name: 'k1', value: 'v1' },
        { name: 'k2', value: 'v2' },
      ],
    },
  },
  {
    id: 'Category Display',
    type: 'CATEGORY_DISPLAY',
    values: { value: 'v2' },
    format: {
      items: [
        { name: 'k1', value: 'v1' },
        { name: 'k2', value: 'v2' },
        { name: 'k3', value: 'v3' },
        { name: 'k4', value: 'v4' },
        { name: 'k5', value: 'v5' },
      ],
    },
  },
  {
    id: 'GPIO Control id',
    type: 'GPIO_CONTROL',
    values: { value: 1 },
    format: {},
  },
  {
    id: 'GPIO Display id',
    type: 'GPIO_DISPLAY',
    values: { value: 0 },
    format: {},
  },
  {
    id: 'Analog Control id',
    type: 'ANALOG_CONTROL',
    values: { value: 23 },
    format: {
      lowerbound: 1,
      upperbound: 100,
    },
  },
  {
    id: 'PWM Display',
    type: 'PWM_DISPLAY',
    values: { value: 'value pwm', period: 20 },
    format: {},
  },
  {
    id: 'PWM Control',
    type: 'PWM_CONTROL',
    values: { value: 0, period: 0 },
    format: {
      lowerbound: 0,
      upperbound: 100,
    },
  },
];
export default DATA_CHANNELS;
