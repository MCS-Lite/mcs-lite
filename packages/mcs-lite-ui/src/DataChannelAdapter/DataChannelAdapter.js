import React, { PropTypes } from 'react';
import styled from 'styled-components';
import R from 'ramda';
import DataChannel from '../DataChannel';

const Wrapper = styled.div`
  width: 100%;
`;

/**
 * type Event = {
 *   type: 'SUBMIT'|'CHANGE'|'CLEAR', // event type
 *   id: string,                      // data channel id
 *   values: {                        // datapoint values
 *     value: string|number,
 *     period?: number,
 *   },
 * }
 * function eventHandler(event: Event): void {}
 */

const NO_DATA_PLACEHOLDER = 'N/A';

class DataChannelAdapter extends React.Component {
  static propTypes = {
    dataChannelProps: PropTypes.shape({
      id: PropTypes.string.isRequired,
      type: PropTypes.oneOf([
        'SWITCH_CONTROL', 'SWITCH_DISPLAY',
        'INTEGER_CONTROL', 'INTEGER_DISPLAY',
        'STRING_CONTROL', 'STRING_DISPLAY',
        'HEX_CONTROL', 'HEX_DISPLAY',
        'FLOAT_DISPLAY', 'FLOAT_CONTROL',
        'GPIO_CONTROL', 'GPIO_DISPLAY',
        'CATEGORY_CONTROL', 'CATEGORY_DISPLAY',
        'ANALOG_CONTROL',
        'PWM_DISPLAY', 'PWM_CONTROL',
      ]).isRequired,
      values: PropTypes.shape({
        value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        period: PropTypes.number,
      }),
      format: PropTypes.shape({
        unit: PropTypes.string,
        items: PropTypes.array,
        lowerbound: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        upperbound: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      }).isRequired,
    }),
    eventHandler: PropTypes.func.isRequired, // (event: object) => void
  }

  // TODO: refactor these handler codes. (maybe hoist)
  switchByType = () => {
    const { dataChannelProps, eventHandler } = this.props;
    const { id, type, format } = dataChannelProps;

    // Remind: API response could be 'null' value.
    const values = R.defaultTo({})(dataChannelProps.values);

    return R.cond([

      /**
       * Switch
       *
       * @author Michael Hsu
       */

      [R.equals('SWITCH_CONTROL'), () =>
        <DataChannel.ControlSwitch
          value={Boolean(values.value)}
          onSubmit={() => eventHandler({
            type: 'SUBMIT',
            id,
            values: { value: !values.value ? 1 : 0 },
          })}
        />,
      ],
      [R.equals('SWITCH_DISPLAY'), () =>
        <DataChannel.DisplayStatus
          value={values.value && values.value ? 1 : 0}
          labels={['OFF', 'ON']}
        />,
      ],

      /**
       * Integer
       *
       * @author Michael Hsu
       */

      [R.equals('INTEGER_CONTROL'), () =>
        <DataChannel.ControlNumber
          placeholder="Integer only"
          unit={format.unit && `單位：${format.unit}`}
          value={R.isNil(values.value) ? '' : parseInt(values.value, 10)}
          onSubmit={() => eventHandler({ type: 'SUBMIT', id, values })}
          onChange={e => eventHandler({ type: 'CHANGE', id, values: { value: e.target.value }})}
          onClear={() => eventHandler({ type: 'CLEAR', id, values: {}})}
        />,
      ],
      [R.equals('INTEGER_DISPLAY'), () =>
        <DataChannel.DisplayUnitValue
          value={R.isNil(values.value) ? NO_DATA_PLACEHOLDER : parseInt(values.value, 10)}
          unit={format.unit}
        />,
      ],

      /**
       * Float
       *
       * @author Michael Hsu
       */

      [R.equals('FLOAT_CONTROL'), () =>
        <DataChannel.ControlNumber
          placeholder="Float only"
          unit={format.unit && `單位：${format.unit}`}
          value={R.isNil(values.value) ? '' : parseFloat(values.value, 10)}
          onSubmit={() => eventHandler({ type: 'SUBMIT', id, values })}
          onChange={e => eventHandler({ type: 'CHANGE', id, values: { value: e.target.value }})}
          onClear={() => eventHandler({ type: 'CLEAR', id, values: {}})}
        />,
      ],
      [R.equals('FLOAT_DISPLAY'), () =>
        <DataChannel.DisplayUnitValue
          value={R.isNil(values.value) ? NO_DATA_PLACEHOLDER : parseFloat(values.value, 10)}
          unit={format.unit}
        />,
      ],

      /**
       * String
       *
       * @author Michael Hsu
       */

      [R.equals('STRING_CONTROL'), () =>
        <DataChannel.ControlString
          placeholder="String only"
          value={R.isNil(values.value) ? '' : values.value}
          onSubmit={() => eventHandler({ type: 'SUBMIT', id, values })}
          onChange={e => eventHandler({ type: 'CHANGE', id, values: { value: e.target.value }})}
          onClear={() => eventHandler({ type: 'CLEAR', id, values: {}})}
        />,
      ],
      [R.equals('STRING_DISPLAY'), () =>
        <DataChannel.DisplayString
          placeholder="String only"
          value={R.isNil(values.value) ? '' : values.value}
        />,
      ],

      /**
       * HEX
       *
       * @author Michael Hsu
       */

      [R.equals('HEX_CONTROL'), () =>
        <DataChannel.ControlString
          placeholder="Hex only"
          value={R.isNil(values.value) ? '' : values.value}
          onSubmit={() => eventHandler({ type: 'SUBMIT', id, values })}
          onChange={e => eventHandler({ type: 'CHANGE', id, values: { value: e.target.value }})}
          onClear={() => eventHandler({ type: 'CLEAR', id, values: {}})}
        />,
      ],
      [R.equals('HEX_DISPLAY'), () =>
        <DataChannel.DisplayString
          placeholder="Hex only"
          value={R.isNil(values.value) ? '' : values.value}
        />,
      ],

      /**
       * GPIO
       *
       * @author Michael Hsu
       */

      [R.equals('GPIO_CONTROL'), () => {
        const labels = ['Low', 'High'];
        const valueMapper = index => labels[index] || NO_DATA_PLACEHOLDER;

        return (
          <DataChannel.ControlRange
            value={R.isNil(values.value) ? '' : values.value}
            labels={labels}
            valueMapper={valueMapper}
            onChange={e => eventHandler({
              type: 'CHANGE',
              id,
              values: { value: e.target.value },
            })}
            onSubmit={() => eventHandler({
              type: 'SUBMIT',
              id,
              values: { value: values.value },
            })}
          />
        );
      }],
      [R.equals('GPIO_DISPLAY'), () =>
        <DataChannel.DisplayStatus
          value={R.isNil(values.value) ? undefined : values.value}
          labels={['Low', 'High']}
        />,
      ],

      /**
       * Category
       *
       * @author Michael Hsu
       */

      [R.equals('CATEGORY_CONTROL'), () => {
        const value = R.isNil(values.value)
          ? undefined
          : R.findIndex(R.propEq('value', values.value))(format.items);
        const valueMapper = index => format.items[index]
          ? format.items[index].value
          : NO_DATA_PLACEHOLDER;

        return (
          <DataChannel.ControlRange
            value={value}
            labels={R.pluck('name')(format.items)}
            valueMapper={valueMapper}
            onChange={e => eventHandler({
              type: 'CHANGE',
              id,
              values: { value: valueMapper(e.target.value) },
            })}
            onSubmit={() => eventHandler({
              type: 'SUBMIT',
              id,
              values: { value: values.value },
            })}
          />
        );
      }],
      [R.equals('CATEGORY_DISPLAY'), () => {
        const value = R.isNil(values.value)
          ? undefined
          : R.findIndex(R.propEq('value', values.value))(format.items);

        return (
          <DataChannel.DisplayStatus
            value={value}
            labels={R.pluck('name')(format.items)}
          />
        );
      }],

      /**
       * Analog (Control only)
       *
       * @author Michael Hsu
       */

      [R.equals('ANALOG_CONTROL'), () => {
        // Hint: index could be empty string. (For Controlled Component usage.)
        const valueMapper = index => R.isNil(index) || R.isEmpty(index)
          ? NO_DATA_PLACEHOLDER
          : index;

        return (
          <DataChannel.ControlRange
            value={R.isNil(values.value) ? '' : values.value}
            valueMapper={valueMapper}
            labels={[
              parseFloat(format.lowerbound, 10),
              parseFloat(format.upperbound, 10),
            ]}
            onChange={e => eventHandler({
              type: 'CHANGE',
              id,
              values: { value: e.target.value },
            })}
            onSubmit={() => eventHandler({
              type: 'SUBMIT',
              id,
              values: { value: values.value },
            })}
          />
        );
      }],

      /**
       * PWM
       *
       * @author Michael Hsu
       */

      [R.equals('PWM_DISPLAY'), () =>
        <DataChannel.DisplayMultipleValue
          items={[
            { name: 'Value', value: R.isNil(values.value) ? NO_DATA_PLACEHOLDER : values.value },
            { name: 'Period', value: R.isNil(values.period) ? NO_DATA_PLACEHOLDER : values.period },
          ]}
        />,
      ],
      [R.equals('PWM_CONTROL'), () => {
        // Hint: index could be empty string. (For Controlled Component usage.)
        const valueMapper = index => R.isNil(index) || R.isEmpty(index)
          ? NO_DATA_PLACEHOLDER
          : index;

        return (
          <Wrapper>
            <DataChannel.ControlPeriod
              value={R.isNil(values.period) ? '' : values.period}
              onSubmit={() => eventHandler({ type: 'SUBMIT', id, values })}
              onChange={e => eventHandler({
                type: 'CHANGE',
                id,
                values: { period: e.target.value, value: values.value },
              })}
              placeholder="Integer only"
            />
            <DataChannel.ControlRange
              value={R.isNil(values.value) ? '' : parseFloat(values.value, 10)}
              valueMapper={valueMapper}
              labels={[
                parseFloat(format.lowerbound, 10),
                parseFloat(format.upperbound, 10),
              ]}
              onChange={e => eventHandler({
                type: 'CHANGE',
                id,
                values: { value: e.target.value, period: values.period },
              })}
              onSubmit={() => eventHandler({
                type: 'SUBMIT',
                id,
                values: { value: values.value, period: values.period },
              })}
            />
          </Wrapper>
        );
      }],
      [R.T, name => <div>{name} NOT SUPPORTED.</div>],
    ])(type);
  }

  render() {
    return this.switchByType();
  }
}

export default DataChannelAdapter;
