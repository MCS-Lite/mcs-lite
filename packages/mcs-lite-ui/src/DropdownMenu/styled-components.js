// @flow
import * as React from 'react';
import styled from 'styled-components';
import Menu from '../Menu/Menu';
import Button from '../Button';

export const StyledButton: React.ComponentType<{
  active: boolean,
}> = styled(Button)`
  display: flex;
  align-items: center;

  svg {
    margin-left: 5px;
    width: 18px;
    height: 18px;
    fill: ${props => props.theme.color.white};
    transition: transform 0.4s cubic-bezier(0.68, -0.55, 0.27, 1.55);
    transform: ${props => (props.active ? 'rotate(-180deg)' : 'initial')};
  }
`;

export const StyledMenu = styled(Menu)`
  border: 1px solid ${props => props.theme.color.grayBase};
`;
