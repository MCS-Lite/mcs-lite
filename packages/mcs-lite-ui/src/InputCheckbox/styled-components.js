// @flow
import styled from 'styled-components';

export const StyledLabel = styled.label`
  display: flex;
  align-items: center;
  cursor: pointer;
  line-height: 0;

  > *:first-child {
    margin-right: 5px;
  }
`;

export default StyledLabel;
