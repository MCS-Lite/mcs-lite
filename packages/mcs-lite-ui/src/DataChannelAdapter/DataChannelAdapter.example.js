import React from 'react';
import styled from 'styled-components';
import { storiesOf, action } from '@kadira/storybook';
import DataChannelCard from '../DataChannelCard';
import DATA_CHANNELS from './API';
import DataChannelAdapter from '.';

const CardWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;

  > * {
    height: initial;
    padding: 8px 16px;
    margin: 4px;
    flex-basis: 100%;
  }

  > [data-width ~= half] {
    flex-grow: 1;
    flex-basis: 40%;
  }
`;

storiesOf('DataChannelAdapter', module)
  .addWithInfo(
    'API',
    `
      ~~~js
      type Event = {
        type: 'SUBMIT'|'CHANGE'|'CLEAR', // event type
        id: string,                      // data channel id
        values: {                        // datapoint values
          value: string|number,
          period?: number,
        },
      }

      function eventHandler(event: Event): void {}
      ~~~
    `,
    () =>
      <CardWrapper>
        {
          DATA_CHANNELS.map(dataChannel => (
            <DataChannelCard
              key={dataChannel.id}
              data-width="half"
              title={dataChannel.type}
              subtitle="Last data point time : 2015-06-12 12:00"
              description="You can input description of controller here. You can input description of You can input description of controller here. You can input description of"
              header={<a href="">Link</a>}
            >
              <DataChannelAdapter
                dataChannelProps={dataChannel}
                eventHandler={action('DataChannelAdapter eventHandler(event: Event)')}
              />
            </DataChannelCard>
          ))
        }
      </CardWrapper>,
    { inline: true },
  );
