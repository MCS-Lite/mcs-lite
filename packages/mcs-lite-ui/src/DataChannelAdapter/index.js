import DataChannelAdapter from './DataChannelAdapter';

export default DataChannelAdapter;
