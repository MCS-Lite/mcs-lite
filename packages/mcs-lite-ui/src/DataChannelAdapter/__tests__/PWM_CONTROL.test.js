import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { theme } from 'mcs-lite-theme';
import DataChannelAdapter from '../DataChannelAdapter';

it('should render PWM_CONTROL correctly with default value to N/A', () => {
  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <DataChannelAdapter
        dataChannelProps={{
          id: 'id',
          type: 'PWM_CONTROL',
          values: {},
          format: {
            lowerbound: 0,
            upperbound: 100,
          },
        }}
        eventHandler={() => {}}
      />
    </ThemeProvider>,
  );

  expect(wrapper.find(DataChannelAdapter)).toMatchSnapshot();
});

it('should render PWM_CONTROL correctly with zero', () => {
  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <DataChannelAdapter
        dataChannelProps={{
          id: 'id',
          type: 'PWM_CONTROL',
          values: { period: 0, value: 0 },
          format: {
            lowerbound: 0,
            upperbound: 100,
          },
        }}
        eventHandler={() => {}}
      />
    </ThemeProvider>,
  );

  expect(wrapper.find(DataChannelAdapter)).toMatchSnapshot();
});

it('should render PWM_CONTROL correctly with max', () => {
  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <DataChannelAdapter
        dataChannelProps={{
          id: 'id',
          type: 'PWM_CONTROL',
          values: { period: 1, value: 100 },
          format: {
            lowerbound: 0,
            upperbound: 100,
          },
        }}
        eventHandler={() => {}}
      />
    </ThemeProvider>,
  );

  expect(wrapper.find(DataChannelAdapter)).toMatchSnapshot();
});
