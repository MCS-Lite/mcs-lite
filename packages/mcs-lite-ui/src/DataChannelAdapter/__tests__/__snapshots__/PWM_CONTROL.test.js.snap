// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PWM_CONTROL correctly with default value to N/A 1`] = `
<DataChannelAdapter
  dataChannelProps={
    Object {
      "format": Object {
        "lowerbound": 0,
        "upperbound": 100,
      },
      "id": "id",
      "type": "PWM_CONTROL",
      "values": Object {},
    }
  }
  eventHandler={[Function]}
>
  <styled.div>
    <div
      className="kdCvHd"
    >
      <ControlPeriod
        onChange={[Function]}
        onSubmit={[Function]}
        placeholder="Integer only"
        value=""
      >
        <styled.div>
          <div
            className="korPeA"
          >
            <label
              htmlFor="input"
            >
              <P
                color="grayBase"
              >
                <Component
                  className="jyQfel"
                  color="grayBase"
                >
                  <p
                    className="jyQfel"
                    color="grayBase"
                  >
                    Period
                  </p>
                </Component>
              </P>
            </label>
            <styled.div>
              <div
                className="fystPB"
              >
                <styled.input
                  kind="primary"
                  onChange={[Function]}
                  placeholder="Integer only"
                  type="number"
                  value=""
                >
                  <input
                    className="cMuAcD"
                    kind="primary"
                    onChange={[Function]}
                    placeholder="Integer only"
                    type="number"
                    value=""
                  />
                </styled.input>
                <Button
                  block={false}
                  component="button"
                  kind="primary"
                  onClick={[Function]}
                >
                  <BaseComponent
                    block={false}
                    className="fqCsyW"
                    component="button"
                    kind="primary"
                    onClick={[Function]}
                  >
                    <button
                      className="fqCsyW"
                      onClick={[Function]}
                    >
                      Ok
                    </button>
                  </BaseComponent>
                </Button>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </ControlPeriod>
      <ControlRange
        labels={
          Array [
            0,
            100,
          ]
        }
        onChange={[Function]}
        onSubmit={[Function]}
        value=""
        valueMapper={[Function]}
      >
        <styled.div>
          <div
            className="ciDLcE"
          >
            <styled.div
              labels={
                Array [
                  0,
                  100,
                ]
              }
            >
              <div
                className="hxnWDy"
              >
                <styled.div>
                  <div
                    className="kRwEkq"
                  >
                    <P
                      color="grayBase"
                    >
                      <Component
                        className="jyQfel"
                        color="grayBase"
                      >
                        <div
                          className="jyQfel"
                          color="grayBase"
                        >
                          0
                        </div>
                      </Component>
                    </P>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="kRwEkq"
                  >
                    <P
                      color="grayBase"
                    >
                      <Component
                        className="jyQfel"
                        color="grayBase"
                      >
                        <div
                          className="jyQfel"
                          color="grayBase"
                        >
                          100
                        </div>
                      </Component>
                    </P>
                  </div>
                </styled.div>
              </div>
            </styled.div>
            <styled.div
              labels={
                Array [
                  0,
                  100,
                ]
              }
            >
              <div
                className="glLEWf"
              >
                <Styled(Component)
                  color="grayBase"
                >
                  <Component
                    className="chvbqM"
                    color="grayBase"
                  >
                    <div
                      className="chvbqM"
                      color="grayBase"
                    >
                      Current value:Â 
                      <Styled(Component)
                        color="primary"
                      >
                        <Component
                          className="eBMqlM"
                          color="primary"
                        >
                          <p
                            className="eBMqlM"
                            color="primary"
                          >
                            N/A
                          </p>
                        </Component>
                      </Styled(Component)>
                    </div>
                  </Component>
                </Styled(Component)>
                <InputRange
                  kind="primary"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  step={1}
                  type="range"
                  value=""
                >
                  <input
                    className="hdhnFx"
                    kind="primary"
                    max={100}
                    min={0}
                    onChange={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    step={1}
                    type="range"
                    value=""
                  />
                </InputRange>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </ControlRange>
    </div>
  </styled.div>
</DataChannelAdapter>
`;

exports[`should render PWM_CONTROL correctly with max 1`] = `
<DataChannelAdapter
  dataChannelProps={
    Object {
      "format": Object {
        "lowerbound": 0,
        "upperbound": 100,
      },
      "id": "id",
      "type": "PWM_CONTROL",
      "values": Object {
        "period": 1,
        "value": 100,
      },
    }
  }
  eventHandler={[Function]}
>
  <styled.div>
    <div
      className="kdCvHd"
    >
      <ControlPeriod
        onChange={[Function]}
        onSubmit={[Function]}
        placeholder="Integer only"
        value={1}
      >
        <styled.div>
          <div
            className="korPeA"
          >
            <label
              htmlFor="input"
            >
              <P
                color="grayBase"
              >
                <Component
                  className="jyQfel"
                  color="grayBase"
                >
                  <p
                    className="jyQfel"
                    color="grayBase"
                  >
                    Period
                  </p>
                </Component>
              </P>
            </label>
            <styled.div>
              <div
                className="fystPB"
              >
                <styled.input
                  kind="primary"
                  onChange={[Function]}
                  placeholder="Integer only"
                  type="number"
                  value={1}
                >
                  <input
                    className="cMuAcD"
                    kind="primary"
                    onChange={[Function]}
                    placeholder="Integer only"
                    type="number"
                    value={1}
                  />
                </styled.input>
                <Button
                  block={false}
                  component="button"
                  kind="primary"
                  onClick={[Function]}
                >
                  <BaseComponent
                    block={false}
                    className="fqCsyW"
                    component="button"
                    kind="primary"
                    onClick={[Function]}
                  >
                    <button
                      className="fqCsyW"
                      onClick={[Function]}
                    >
                      Ok
                    </button>
                  </BaseComponent>
                </Button>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </ControlPeriod>
      <ControlRange
        labels={
          Array [
            0,
            100,
          ]
        }
        onChange={[Function]}
        onSubmit={[Function]}
        value={100}
        valueMapper={[Function]}
      >
        <styled.div>
          <div
            className="ciDLcE"
          >
            <styled.div
              labels={
                Array [
                  0,
                  100,
                ]
              }
            >
              <div
                className="hxnWDy"
              >
                <styled.div>
                  <div
                    className="kRwEkq"
                  >
                    <P
                      color="grayBase"
                    >
                      <Component
                        className="jyQfel"
                        color="grayBase"
                      >
                        <div
                          className="jyQfel"
                          color="grayBase"
                        >
                          0
                        </div>
                      </Component>
                    </P>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="kRwEkq"
                  >
                    <P
                      color="grayBase"
                    >
                      <Component
                        className="jyQfel"
                        color="grayBase"
                      >
                        <div
                          className="jyQfel"
                          color="grayBase"
                        >
                          100
                        </div>
                      </Component>
                    </P>
                  </div>
                </styled.div>
              </div>
            </styled.div>
            <styled.div
              labels={
                Array [
                  0,
                  100,
                ]
              }
            >
              <div
                className="glLEWf"
              >
                <Styled(Component)
                  color="grayBase"
                >
                  <Component
                    className="chvbqM"
                    color="grayBase"
                  >
                    <div
                      className="chvbqM"
                      color="grayBase"
                    >
                      Current value:Â 
                      <Styled(Component)
                        color="primary"
                      >
                        <Component
                          className="eBMqlM"
                          color="primary"
                        >
                          <div
                            className="eBMqlM"
                            color="primary"
                          >
                            100
                          </div>
                        </Component>
                      </Styled(Component)>
                    </div>
                  </Component>
                </Styled(Component)>
                <InputRange
                  kind="primary"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  step={1}
                  type="range"
                  value={100}
                >
                  <input
                    className="hdhnFx"
                    kind="primary"
                    max={100}
                    min={0}
                    onChange={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    step={1}
                    type="range"
                    value={100}
                  />
                </InputRange>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </ControlRange>
    </div>
  </styled.div>
</DataChannelAdapter>
`;

exports[`should render PWM_CONTROL correctly with zero 1`] = `
<DataChannelAdapter
  dataChannelProps={
    Object {
      "format": Object {
        "lowerbound": 0,
        "upperbound": 100,
      },
      "id": "id",
      "type": "PWM_CONTROL",
      "values": Object {
        "period": 0,
        "value": 0,
      },
    }
  }
  eventHandler={[Function]}
>
  <styled.div>
    <div
      className="kdCvHd"
    >
      <ControlPeriod
        onChange={[Function]}
        onSubmit={[Function]}
        placeholder="Integer only"
        value={0}
      >
        <styled.div>
          <div
            className="korPeA"
          >
            <label
              htmlFor="input"
            >
              <P
                color="grayBase"
              >
                <Component
                  className="jyQfel"
                  color="grayBase"
                >
                  <p
                    className="jyQfel"
                    color="grayBase"
                  >
                    Period
                  </p>
                </Component>
              </P>
            </label>
            <styled.div>
              <div
                className="fystPB"
              >
                <styled.input
                  kind="primary"
                  onChange={[Function]}
                  placeholder="Integer only"
                  type="number"
                  value={0}
                >
                  <input
                    className="cMuAcD"
                    kind="primary"
                    onChange={[Function]}
                    placeholder="Integer only"
                    type="number"
                    value={0}
                  />
                </styled.input>
                <Button
                  block={false}
                  component="button"
                  kind="primary"
                  onClick={[Function]}
                >
                  <BaseComponent
                    block={false}
                    className="fqCsyW"
                    component="button"
                    kind="primary"
                    onClick={[Function]}
                  >
                    <button
                      className="fqCsyW"
                      onClick={[Function]}
                    >
                      Ok
                    </button>
                  </BaseComponent>
                </Button>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </ControlPeriod>
      <ControlRange
        labels={
          Array [
            0,
            100,
          ]
        }
        onChange={[Function]}
        onSubmit={[Function]}
        value={0}
        valueMapper={[Function]}
      >
        <styled.div>
          <div
            className="ciDLcE"
          >
            <styled.div
              labels={
                Array [
                  0,
                  100,
                ]
              }
            >
              <div
                className="hxnWDy"
              >
                <styled.div>
                  <div
                    className="kRwEkq"
                  >
                    <P
                      color="grayBase"
                    >
                      <Component
                        className="jyQfel"
                        color="grayBase"
                      >
                        <div
                          className="jyQfel"
                          color="grayBase"
                        >
                          0
                        </div>
                      </Component>
                    </P>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="kRwEkq"
                  >
                    <P
                      color="grayBase"
                    >
                      <Component
                        className="jyQfel"
                        color="grayBase"
                      >
                        <div
                          className="jyQfel"
                          color="grayBase"
                        >
                          100
                        </div>
                      </Component>
                    </P>
                  </div>
                </styled.div>
              </div>
            </styled.div>
            <styled.div
              labels={
                Array [
                  0,
                  100,
                ]
              }
            >
              <div
                className="glLEWf"
              >
                <Styled(Component)
                  color="grayBase"
                >
                  <Component
                    className="chvbqM"
                    color="grayBase"
                  >
                    <div
                      className="chvbqM"
                      color="grayBase"
                    >
                      Current value:Â 
                      <Styled(Component)
                        color="primary"
                      >
                        <Component
                          className="eBMqlM"
                          color="primary"
                        >
                          <div
                            className="eBMqlM"
                            color="primary"
                          >
                            0
                          </div>
                        </Component>
                      </Styled(Component)>
                    </div>
                  </Component>
                </Styled(Component)>
                <InputRange
                  kind="primary"
                  max={100}
                  min={0}
                  onChange={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  step={1}
                  type="range"
                  value={0}
                >
                  <input
                    className="hdhnFx"
                    kind="primary"
                    max={100}
                    min={0}
                    onChange={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    step={1}
                    type="range"
                    value={0}
                  />
                </InputRange>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </ControlRange>
    </div>
  </styled.div>
</DataChannelAdapter>
`;
